components/inspectit-ocelot-configurationserver/src/main/java/rocks/inspectit/ocelot/agentconfiguration/AgentConfigurationReloadTask.java

```diff
@@ -172,7 +169,7 @@ private Object loadAndMergeYaml(AbstractFileAccessor fileAccessor, Object toMerg
         String src = fileAccessor.readConfigurationFile(path).orElse("");
 
         try {
-            Object loadedYaml = yaml.load(src);
+            Map<String, Object> loadedYaml = yaml.load(src);
```

Comment:
This change leads to yaml.load throwing an Exception when it encounters any Yaml that does not represent a Map. This means that one invalid file containing only a String or a List being parsed first can not lead to all other files' contents being ignored anymore, because now any invalid file like that will simply be logged as invalid and skipped.